batch_job/src/work_queue_factory.c

```diff
@@ -1331,6 +1335,30 @@ int main(int argc, char *argv[])
 		return 1;
 	}
 
+	if(wrapper_input) {
+		struct jx *item;
+		for(void *i = NULL; (item = jx_iterate_array(wrapper_inputs, &i));) {
+			const char *value = item->u.string_value;
+			const char *file_at_scratch_dir = string_format("%s/%s", scratch_dir, path_basename(value));
+			int64_t result = 0; 
```

Comment:
Save some space and use `copy_direntry` here, which handles both cases of file and dir...

Reply:
Done.